data-processing/common/commands/fetch_s2_data.py

```diff
@@ -99,13 +102,28 @@ def process(self, item: ArxivId) -> Iterator[S2Metadata]:
                 data = resp.json()
                 references = []
                 for reference_data in data["references"]:
+
+                    if reference_data["paperId"] is None:
```

Comment:
Mostly a curiosity thing - why have this case instead of having just `if reference_data["paperId"]: do stuff`?

Reply:
oh lol, yeah it's just my old habit of throwing out unwanted stuff in the beginning, then everything afterwards can just be the big chunk of code where I work on stuff I actually want.

```
if (unwanted condition):
  skip this one

# huge chunk of code specifically for ones I want
# do this
# do that
# etc...
```

Whereas I suppose this would be cleaner:
```
if (wanted condition):
  # do this
  # do that
  # etc...
```

Will change!!